Anagram.java

```diff
@@ -28,22 +30,53 @@ public static void main(String args[]) {
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
+		str1 = preProcess(str1);
+		str2 = preProcess(str2);
+		str1 = str1.replace(" ", "");
+		str2 = str2.replace(" ", "");
+
+		if (str1.length() != str2.length()) {
+			return false;
+		}
+		while (str1.length() > 0 && str2.length() > 0) {
+			int randomIndex = (int) (Math.random() * str1.length());
```

Comment:
Using a random generated index is **ok and works**, but why not just "walk" from the start to the end of the Strings? 🤷🏻‍♂️